clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java

```diff
@@ -371,6 +375,35 @@ public synchronized AbstractStorageEngine openStore(
     return engine;
   }
 
+  public synchronized void checkWhetherStoragePartitionsShouldBeKeptOrNot(SafeHelixManager manager) {
+    if (manager == null) {
+      return;
+    }
+    for (AbstractStorageEngine storageEngine: getStorageEngineRepository().getAllLocalStorageEngines()) {
+      String storeName = storageEngine.getStoreVersionName();
+      Set<Integer> storageEnginePartitionIds = new HashSet<>(storageEngine.getPartitionIds());
```

Comment:
There is one edge case that we need to be aware of here.  Venice internally assigns a metadata partition to the storage engine.  This metadata partition exists on every Venice replica.  This metadata partition will not be part of the ideal state.  If you look at AbstractStorageEngine.METADATA_PARTITION_ID you can use this to screen dropping that particular partition (as it won't be in the ideal state).